### ISSUE TITLE
Error installing open3d on a raspberry Pi 4

### ISSUE THREAD

**Issue by LucasSebille** [isl-org/Open3D#4240]
**Describe the bug**
I want to install open3d on a raspeberry PI 4 using the code that you can see below.

**To Reproduce**
```
sudo apt-get update -y
sudo apt-get install -y apt-utils build-essential git cmake
sudo apt-get install -y python3 python3-dev python3-pip
sudo apt-get install -y xorg-dev libglu1-mesa-dev
sudo apt-get install -y libblas-dev liblapack-dev liblapacke-dev
sudo apt-get install -y libsdl2-dev libc++-7-dev libc++abi-7-dev libxi-dev
sudo apt-get install -y clang-7

sudo apt-get install -y python3-virtualenv ccache
```
# Clone
```
git clone --recursive https://github.com/isl-org/Open3D
cd Open3D
git submodule update --init --recursive
mkdir build
cd build
```

# Configure
```
# > Set -DBUILD_CUDA_MODULE=ON if CUDA is available (e.g. on Nvidia Jetson)
# > Set -DBUILD_GUI=ON if OpenGL is available (e.g. on Nvidia Jetson)
# > We don't support TensorFlow and PyTorch on ARM officially
cmake \
    -DCMAKE_BUILD_TYPE=Release \
    -DBUILD_SHARED_LIBS=ON \
    -DBUILD_CUDA_MODULE=OFF \
    -DBUILD_GUI=OFF \
    -DBUILD_TENSORFLOW_OPS=OFF \
    -DBUILD_PYTORCH_OPS=OFF \
    -DBUILD_ISPC_MODULE=OFF \
    -DBUILD_UNIT_TESTS=ON \
    -DCMAKE_INSTALL_PREFIX=~/open3d_install \
    ..
```
# Build C++ library
```
make -j$(nproc)
```

# Results

When trying the last line, i get this :
```
[  1%] Performing build step for 'ext_embree'
[  1%] Built target ext_mkl
[  3%] Built target ext_tbb
[  3%] Built target ext_faiss
[  4%] Built target ext_msgpack-c
[  5%] Built target ext_googletest
[  6%] Built target ext_zeromq
[  7%] Built target ext_mkl_include
[  0%] Building CXX object common/sys/CMakeFiles/sys.dir/sysinfo.cpp.o
[  1%] Building CXX object common/sys/CMakeFiles/sys.dir/alloc.cpp.o
c++: error: unrecognized command line option ‘-mavx’; did you mean ‘-marm’?
make[5]: *** [common/sys/CMakeFiles/sys.dir/build.make:76: common/sys/CMakeFiles/sys.dir/sysinfo.cpp.o] Error 1
make[5]: *** Waiting for unfinished jobs....
[  7%] Creating directories for 'ext_poisson'
c++: error: unrecognized command line option ‘-mavx’; did you mean ‘-marm’?
make[5]: *** [common/sys/CMakeFiles/sys.dir/build.make:90: common/sys/CMakeFiles/sys.dir/alloc.cpp.o] Error 1
make[4]: *** [CMakeFiles/Makefile2:239: common/sys/CMakeFiles/sys.dir/all] Error 2
make[3]: *** [Makefile:156: all] Error 2
make[2]: *** [CMakeFiles/ext_embree.dir/build.make:86: embree/src/ext_embree-stamp/ext_embree-build] Error 2
make[1]: *** [CMakeFiles/Makefile2:1207: CMakeFiles/ext_embree.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  7%] Creating directories for 'ext_zlib'
[  7%] Built target open3d_downloads
[  7%] Performing download step (download, verify and extract) for 'ext_poisson'
-- Downloading...
   dst='/home/pi/Open3D/3rdparty_downloads/poisson/fd273ea8c77a36973d6565a495c9969ccfb12d3b.tar.gz'
   timeout='none'
   inactivity timeout='none'
-- Using src='https://github.com/isl-org/Open3D-PoissonRecon/archive/fd273ea8c77a36973d6565a495c9969ccfb12d3b.tar.gz'
[  7%] Performing download step (download, verify and extract) for 'ext_zlib'
-- Downloading...
   dst='/home/pi/Open3D/3rdparty_downloads/zlib/v1.2.11.tar.gz'
   timeout='none'
   inactivity timeout='none'
-- Using src='https://github.com/madler/zlib/archive/refs/tags/v1.2.11.tar.gz'
-- [download 100% complete]
-- [download 100% complete]
-- verifying file...
       file='/home/pi/Open3D/3rdparty_downloads/zlib/v1.2.11.tar.gz'
-- Downloading... done
-- extracting...
     src='/home/pi/Open3D/3rdparty_downloads/zlib/v1.2.11.tar.gz'
     dst='/home/pi/Open3D/build/zlib/src/ext_zlib'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[  7%] No update step for 'ext_zlib'
[  7%] No patch step for 'ext_zlib'
[  7%] Performing configure step for 'ext_zlib'
-- verifying file...
       file='/home/pi/Open3D/3rdparty_downloads/poisson/fd273ea8c77a36973d6565a495c9969ccfb12d3b.tar.gz'
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Downloading... done
-- extracting...
     src='/home/pi/Open3D/3rdparty_downloads/poisson/fd273ea8c77a36973d6565a495c9969ccfb12d3b.tar.gz'
     dst='/home/pi/Open3D/build/poisson/src/ext_poisson/PoissonRecon'
-- extracting... [tar xfz]
-- extracting... [analysis]
-- extracting... [rename]
-- extracting... [clean up]
-- extracting... done
[  7%] No update step for 'ext_poisson'
[  7%] No patch step for 'ext_poisson'
-- The C compiler identification is GNU 8.3.0
[  8%] No configure step for 'ext_poisson'
-- Detecting C compiler ABI info
[  8%] No build step for 'ext_poisson'
[  8%] No install step for 'ext_poisson'
[  8%] Completed 'ext_poisson'
[  8%] Built target ext_poisson
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of off64_t
-- Check size of off64_t - done
-- Looking for fseeko
-- Looking for fseeko - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Renaming
--     /home/pi/Open3D/build/zlib/src/ext_zlib/zconf.h
-- to 'zconf.h.included' because this file is included with zlib
-- but CMake generates it automatically in the build directory.
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_CUDA_COMPILER
    CMAKE_CXX_COMPILER


-- Build files have been written to: /home/pi/Open3D/build/zlib/src/ext_zlib-build
[  9%] Performing build step for 'ext_zlib'
[ 10%] Building C object CMakeFiles/zlib.dir/adler32.o
[ 10%] Building C object CMakeFiles/zlibstatic.dir/compress.o
[ 10%] Building C object CMakeFiles/zlib.dir/compress.o
[ 10%] Building C object CMakeFiles/zlibstatic.dir/adler32.o
[ 12%] Building C object CMakeFiles/zlib.dir/crc32.o
[ 15%] Building C object CMakeFiles/zlibstatic.dir/crc32.o
[ 20%] Building C object CMakeFiles/zlibstatic.dir/deflate.o
[ 20%] Building C object CMakeFiles/zlib.dir/deflate.o
[ 22%] Building C object CMakeFiles/zlib.dir/gzclose.o
[ 25%] Building C object CMakeFiles/zlibstatic.dir/gzclose.o
[ 27%] Building C object CMakeFiles/zlibstatic.dir/gzlib.o
[ 30%] Building C object CMakeFiles/zlibstatic.dir/gzread.o
[ 32%] Building C object CMakeFiles/zlibstatic.dir/gzwrite.o
[ 35%] Building C object CMakeFiles/zlibstatic.dir/inflate.o
[ 37%] Building C object CMakeFiles/zlibstatic.dir/infback.o
[ 40%] Building C object CMakeFiles/zlib.dir/gzlib.o
[ 42%] Building C object CMakeFiles/zlibstatic.dir/inftrees.o
[ 45%] Building C object CMakeFiles/zlibstatic.dir/inffast.o
[ 47%] Building C object CMakeFiles/zlib.dir/gzread.o
[ 50%] Building C object CMakeFiles/zlibstatic.dir/trees.o
[ 52%] Building C object CMakeFiles/zlibstatic.dir/uncompr.o
[ 55%] Building C object CMakeFiles/zlibstatic.dir/zutil.o
[ 57%] Building C object CMakeFiles/zlib.dir/gzwrite.o
[ 60%] Building C object CMakeFiles/zlib.dir/inflate.o
[ 62%] Building C object CMakeFiles/zlib.dir/infback.o
[ 65%] Linking C static library libz.a
[ 67%] Building C object CMakeFiles/zlib.dir/inftrees.o
[ 67%] Built target zlibstatic
[ 70%] Building C object CMakeFiles/zlib.dir/inffast.o
[ 72%] Building C object CMakeFiles/zlib.dir/trees.o
[ 75%] Building C object CMakeFiles/zlib.dir/uncompr.o
[ 77%] Building C object CMakeFiles/zlib.dir/zutil.o
[ 80%] Linking C shared library libz.so
[ 80%] Built target zlib
[ 82%] Building C object CMakeFiles/example.dir/test/example.o
[ 85%] Building C object CMakeFiles/example64.dir/test/example.o
[ 87%] Building C object CMakeFiles/minigzip.dir/test/minigzip.o
[ 90%] Building C object CMakeFiles/minigzip64.dir/test/minigzip.o
[ 92%] Linking C executable minigzip
[ 95%] Linking C executable minigzip64
[ 95%] Built target minigzip
[ 95%] Built target minigzip64
[ 97%] Linking C executable example
[100%] Linking C executable example64
[100%] Built target example
[100%] Built target example64
[  9%] Performing install step for 'ext_zlib'
Consolidate compiler generated dependencies of target zlib
Consolidate compiler generated dependencies of target zlibstatic
[ 40%] Built target zlib
[ 80%] Built target zlibstatic
Consolidate compiler generated dependencies of target example
Consolidate compiler generated dependencies of target example64
Consolidate compiler generated dependencies of target minigzip
Consolidate compiler generated dependencies of target minigzip64
[ 85%] Built target example
[ 90%] Built target example64
[ 95%] Built target minigzip
[100%] Built target minigzip64
Install the project...
-- Install configuration: "Release"
-- Installing: /home/pi/Open3D/build/zlib/lib/libz.so.1.2.11
-- Installing: /home/pi/Open3D/build/zlib/lib/libz.so.1
-- Installing: /home/pi/Open3D/build/zlib/lib/libz.so
-- Installing: /home/pi/Open3D/build/zlib/lib/libz.a
-- Installing: /home/pi/Open3D/build/zlib/include/zconf.h
-- Installing: /home/pi/Open3D/build/zlib/include/zlib.h
-- Installing: /home/pi/Open3D/build/zlib/share/man/man3/zlib.3
-- Installing: /home/pi/Open3D/build/zlib/share/pkgconfig/zlib.pc
[  9%] Completed 'ext_zlib'
[  9%] Built target ext_zlib
make: *** [Makefile:136: all] Error 2
```

**Comment by AlexSkyline** [comment]
Hello
What is the version of your raspberry Pi 4? (especially the ram)
Try to restart the compilation with the command "make -j1"

**Comment by LucasSebille** [comment]
It's a raspberry Pi 4 Computer Model BGB RAM

**Comment by LucasSebille** [comment]
It's a raspberry Pi 4 Computer Model B 4GB RAM *

**Comment by LucasSebille** [comment]
I tried with make -j1 but i got the same mystake

**Comment by AlexSkyline** [comment]
even with the 8go version i disable the boot on the graphical interface and i got errors if i did not limit to make -j1 the last compilation you started it from scratch?

**Comment by chnmrc** [comment]
result of
`make VERBOSE=1`

**Comment by yxlao** [comment]
Please add a line in `3rdparty/embree/embree.cmake` and run the configure and build again to get the printout of the flags.

```diff
diff --git a/3rdparty/embree/embree.cmake b/3rdparty/embree/embree.cmake
index 7f3475510..589a68b1b 100644
--- a/3rdparty/embree/embree.cmake
+++ b/3rdparty/embree/embree.cmake
@@ -39,7 +39,7 @@ else() # Linux(x86) and WIN32
     )
 endif()
 
-
+message(FATAL_ERROR "${LINUX_AARCH64} ${ISA_ARGS}")
 
 if(WIN32)
     set(WIN_CMAKE_ARGS "-DCMAKE_CXX_FLAGS_DEBUG=$<IF:$<BOOL:${STATIC_WINDOWS_RUNTIME}>,/MTd,/MDd> ${CMAKE_CXX_FLAGS_DEBUG_INIT}"
```

On x86-64, I get:
```
CMake Error at 3rdparty/embree/embree.cmake:42 (message):
   -DEMBREE_ISA_AVX=ON;-DEMBREE_ISA_AVX2=ON;-DEMBREE_ISA_AVX512=OFF;-DEMBREE_ISA_SSE2=OFF;-DEMBREE_ISA_SSE42=OFF
```

@AlexSkyline Let me know what you get on the Pi 4.

**Comment by yxlao** [comment]
This was addressed in #4428. #4428 fixes both Apple Silicon and Linux ARM64 Embree build.
